tools/ci_build/build.py

```diff
@@ -783,6 +797,8 @@ def main():
               run_onnx_tests(build_dir, configs, onnx_test_data_dir, None, False, 1)
             elif args.use_ngraph:
               run_onnx_tests(build_dir, configs, onnx_test_data_dir, 'ngraph', True, 1)
+            if args.use_openvino:
```

Comment:
i think there is something wrong with the control flow here. (observe the ngraph build fails because it's trying to run with mkldnn options)

can you change line 800 to an elif  and remove the openvino check at line 787?
the control flow should be such that we only run onnx tests with a single provider. 
i.e. with openvino build option, we run onnx_test_runner with -e openvino  and no other tests should explicitly with other providers.

Reply:
Hey George, 
When i was merging code from upstream, i think this old change got carried forward somehow. 
I will make this change and push it to the repo